app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt

```diff
@@ -284,14 +256,9 @@ class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListe
         directionsButton.setOnClickListener {
             bottomSheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
 
-            //Get the building that the user clicked on
-            var selectedBuilding : Building? = null
-            for (campus in viewModel.getCampuses()) {
-                for (building in campus.getBuildings()) {
-                    if (p.tag.toString() == building.getName())
-                        selectedBuilding  = building
-                }
-            }
+            //Get the building object from the polygon that the user clicked on
+            var selectedBuilding : Building? = viewModel.findBuildingByPolygonTag(p.tag.toString())
+
```

Comment:
why add this extra space?

Reply:
I've removed the extra space in my new commit.